apis/v1beta1/validation/httproute.go

```diff
@@ -338,10 +365,6 @@ func hasExactlyOnePrefixMatch(matches []gatewayv1b1.HTTPRouteMatch) bool {
 	if len(matches) != 1 || matches[0].Path == nil {
 		return false
 	}
-	pathMatchType := matches[0].Path.Type
-	if *pathMatchType != gatewayv1b1.PathMatchPathPrefix {
-		return false
-	}
 
-	return true
+	return *matches[0].Path.Type == gatewayv1b1.PathMatchPathPrefix
```

Comment:
This feels not very safe and also not relevant to the rest of the PR, maybe revert this change?